### PR TITLE
Add UnwrapAsync static extensions for Option and Result

### PR DESCRIPTION
**Pull request by b4n4n4j03**
Option: Unwrap for Task<Option<T>>
Result: Unwrap for Task<Result<TOk, TErr>>

### CODE REVIEW

**Comment by phkiener** [comment]
We've talked about this in person, but just so that this PR isn't just hanging by: Yeehaw for async, but I feel like the other variants of Unwrap should be asyncified as well 😉

**Comment by b4n4n4j03** [comment]
@phkiener as discussed, i added the other methods for async support of Unwrap methods.

File: Galaxus.Functional/(Result)/ResultExtensions.cs [hunk 1]
```diff
@@ -31,7 +31,8 @@ public static class ResultExtensions
         /// <typeparam name="TOkFrom">The current type of <b>Ok</b>. This type must derive from <typeparamref name="TOkTo"/>.</typeparam>
         /// <typeparam name="TOkTo">The type to map <b>Ok</b> to.</typeparam>
         /// <typeparam name="TErr">The type of <b>Err</b> which will remain untouched.</typeparam>
-        public static Result<TOkTo, TErr> Map<TOkFrom, TOkTo, TErr>(this Result<TOkFrom, TErr> self) where TOkFrom : TOkTo
+        public static Result<TOkTo, TErr> Map<TOkFrom, TOkTo, TErr>(this Result<TOkFrom, TErr> self)
+            where TOkFrom : TOkTo
```
Comment (phkiener) [inline Galaxus.Functional/(Result)/ResultExtensions.cs:35]: Yeah, autoformatting I guess. We definitely need to have at least a style guide or something...

The where-clauses are alright, I like them better when they're clearly visible, but there's parameter wrapping for SelectOk which is... "not ok" :D

File: Galaxus.Functional.Tests/(Option)/OptionExtensions/UnwrapAsyncTests.cs [hunk 2]
```diff
@@ -0,0 +1,453 @@
...
+using NUnit.Framework;
+
+namespace Galaxus.Functional.Tests.OptionExtensions
+{
+    public class UnwrapAsyncTests
+    {
+        [Test]
+        public async Task UnwrapAsync_WhenOptionIsSome_ThenCorrectValueReturned()
+        {
+            var some = Task.FromResult("hello".ToOption());
```
Comment (phkiener) [inline Galaxus.Functional.Tests/(Option)/OptionExtensions/UnwrapAsyncTests.cs:12]: I know that there's never been much of an Arrange-Act-Assert in the tests here... you think we should start doing that?

I am reading these tests and feel like I'm getting stomped by them. Funcs and Tasks always make tests more unpleasant to look at in my book, we can maybe "soften the impact" here.

What'd'ya think?

File: Galaxus.Functional.Tests/(Result)/TExtensionsForResultTest.cs [hunk 3]
```diff
@@ -158,5 +162,437 @@ public async Task OrElseAsync_SynchronousContinuation_ReturnsSuccessTask()
             await resultTask;
             Assert.AreEqual(expected: TaskStatus.RanToCompletion, actual: resultTask.Status);
         }
+
+
+        [Test]
```
Comment (phkiener) [inline Galaxus.Functional.Tests/(Result)/TExtensionsForResultTest.cs:167]: Now I'm curious - the UnwrapAsync for Option was deserving of its own test file, but UnwrapAsync for Result is not? :P

File: Galaxus.Functional/(Result)/ResultExtensions.cs (hunk 1 above)
Comment (b4n4n4j03) [reply Galaxus.Functional/(Result)/ResultExtensions.cs:35]: Ok I see... I think a style guide or something would not be bad..

File: Galaxus.Functional.Tests/(Result)/TExtensionsForResultTest.cs (hunk 3 above)
Comment (b4n4n4j03) [reply Galaxus.Functional.Tests/(Result)/TExtensionsForResultTest.cs:167]: Ok I splitted now all Tests von TExtensionsForResultTest.cs in separate files. But honestly, there is kind of a mess in this library and we should really invest some time in a clear structure and some style guides.

**Review by phkiener — APPROVED** [review]

**Pull request merged**